File: pdftract-java/src/main/java/com/jedarden/pdftract/Span.java
package com.jedarden.pdftract;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * A text span with font and position information.
 */
public record Span(
    @JsonProperty("text") String text,
    @JsonProperty("font") String font,
    @JsonProperty("size") Double size,
    @JsonProperty("bbox") List<Double> bbox
) {
    public Span {
        bbox = bbox != null ? bbox : List.of();
    }
}

File: pdftract-dotnet/src/Pdftract/Models/Span.cs
using System.Text.Json.Serialization;

namespace Pdftract.Models;

/// <summary>
/// Represents a text span with font and position information.
/// </summary>
public record Span
{
    [JsonPropertyName("text")]
    public required string Text { get; init; }

    [JsonPropertyName("bbox")]
    public required double[] Bbox { get; init; }

    [JsonPropertyName("font")]
    public required string Font { get; init; }

    [JsonPropertyName("size")]
    public required double Size { get; init; }

    [JsonPropertyName("confidence")]
    public double? Confidence { get; init; }
}
